datasafe-encryption/datasafe-encryption-impl/src/main/java/de/adorsys/datasafe/encrypiton/impl/keystore/KeyStoreServiceImpl.java

```diff
@@ -62,17 +71,21 @@ public KeyStore createKeyStore(KeyStoreAuth keyStoreAuth,
         // TODO, hier also statt der StoreID nun das
         String serverKeyPairAliasPrefix = UUID.randomUUID().toString();
         log.debug("keystoreid = {}", serverKeyPairAliasPrefix);
-        KeyStoreGenerator keyStoreGenerator = KeyStoreGenerator.builder()
-                .keyCreationConfig(keyConfig)
-                .keyStoreConfig(config)
-                .serverKeyPairAliasPrefix(serverKeyPairAliasPrefix)
-                .readKeyPassword(keyStoreAuth.getReadKeyPassword())
-                .secretKeys(secretKeys)
+
+        Juggler juggler = DaggerJuggler.builder().build();
+        EncryptingKeyCreationCfg encConf = keyConfig.getEncrypting();
+        SigningKeyCreationCfg signConf = keyConfig.getSigning();
+        KeySetTemplate template = KeySetTemplate.builder()
+                .generatedEncryptionKeys(Encrypting.with().algo(encConf.getAlgo()).sigAlgo(encConf.getSigAlgo()).keySize(encConf.getSize()).prefix("ENC").password(() -> keyStoreAuth.getReadKeyPassword().getValue()).build().repeat(keyConfig.getEncKeyNumber()))
+                .generatedSigningKeys(Signing.with().algo(signConf.getAlgo()).sigAlgo(signConf.getSigAlgo()).keySize(signConf.getSize()).alias("SIGN-1").password(() -> keyStoreAuth.getReadKeyPassword().getValue()).build().repeat(keyConfig.getSignKeyNumber()))
+                .generatedSecretKeys(secretKeys.keySet().stream().map(it -> Secret.with().prefix(it.getValue()).build()).collect(Collectors.toList()))
                 .build();
+        KeySet keySet = juggler.generateKeys().fromTemplate(template);
 
-        KeyStore userKeyStore = keyStoreGenerator.generate();
+        juggler.withConfig(config);
```

Comment:
This is dangling statement - it should have created new instance of some class for immutability